code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -374,12 +373,32 @@
 /datum/dynamic_ruleset/roundstart/revs/execute()
 	revolution = new()
 	for(var/datum/mind/M in assigned)
-		var/datum/antagonist/rev/head/new_head = new antag_datum()
-		new_head.give_flash = TRUE
-		new_head.give_hud = TRUE
-		new_head.remove_clumsy = TRUE
-		M.add_antag_datum(new_head,revolution)
-		GLOB.pre_setup_antags -= M
+		if(check_eligible(M))
+			var/datum/antagonist/rev/head/new_head = new antag_datum()
+			new_head.give_flash = TRUE
+			new_head.give_hud = TRUE
+			new_head.remove_clumsy = TRUE
+			M.add_antag_datum(new_head,revolution)
+			GLOB.pre_setup_antags -= M
+		else if(candidates.len)
+			GLOB.pre_setup_antags -= M
+			assigned -= M
+			log_game("Dynamic: [ruletype] [name] discarded [M.name] due to ineligibility and picked a new head revolutionary.")
+			var/mob/newrev
+			for(var/i in 1 to candidates.len)
+				newrev = pick_n_take(candidates)	// Still picking from roundstart candidates
+				if(check_eligible(newrev.mind))
+					break
+			assigned += newrev.mind
+			newrev.mind.restricted_roles = restricted_roles
+			newrev.mind.special_role = antag_flag
+			var/datum/antagonist/rev/head/new_head = new antag_datum()
+			new_head.give_flash = TRUE
+			new_head.give_hud = TRUE
+			new_head.remove_clumsy = TRUE
+			newrev.mind.add_antag_datum(new_head,revolution)
```

Comment:
This will not work as candidates is already an empty list and if it wasn't it would be probably just list of nulls.

Reply:
And anyways just having one head is enough as then the system can "promote" people.

Reply:
Ah, hm yes you are right. Fuck it then, thanks.

Reply:
If nothing else if there are not enough valid revheads it should refund the ruleset and then kill the processing.